File: Coding Exercises/Python/Day55/is_abundant.py
def is_abundant(num):
    if num <= 0:
        raise ValueError("Input number must be a positive integer")

    sum_of_divisors = 1

    for i in range(2, int(num**0.5) + 1):
        if num % i == 0:
            sum_of_divisors += i
            if i != num // i:
                sum_of_divisors += num // i

    return sum_of_divisors > num

File: Coding Exercises/TypeScript/Day55/isAbundant.ts
function isAbundant(num: number): boolean {
  if (num <= 0) {
      throw new Error("Input number must be a positive integer");
  }

  let sumOfDivisors = 1;

  for (let i = 2; i * i <= num; i++) {
      if (num % i === 0) {
          sumOfDivisors += i;
          if (i !== num / i) {
              sumOfDivisors += num / i;
          }
      }
  }

  return sumOfDivisors > num;
}
